### ISSUE TITLE
C-PAC 1.0.1 - 'Test Configuration' in pipeline configuration editor fails when using S3 participant list

### ISSUE THREAD

**Issue by jpellman** [FCP-INDI/C-PAC#636]
Using a participant list with the configuration tester where NifTIs are stored in S3 produces the following error:

```
Checking subject list: /home/jpellman/nki-rs-preproc/settings/CPAC_subject_list_NKI-RS.yml...
Traceback (most recent call last):
  File "/home2/jpellman/miniconda/envs/cpac101/lib/python2.7/site-packages/CPAC/GUI/interface/windows/config_window.py", line 570, in testConfig
    sub_flg = self.test_sublist(sublist)
  File "/home2/jpellman/miniconda/envs/cpac101/lib/python2.7/site-packages/CPAC/GUI/interface/windows/config_window.py", line 462, in test_sublist
    if func_file.lower().startswith(s3_str):
AttributeError: 'NoneType' object has no attribute 'lower'
```

It looks like it might be related to https://github.com/FCP-INDI/C-PAC/issues/633.  The participant list in question is [here](https://github.com/FCP-INDI/nki-rs-preproc/blob/master/settings/CPAC_subject_list_NKI-RS.yml).  `scan_params` is set to null for every participant, which would cause func_file to be interpreted as a 'None'.

**Comment by amadeuskanaan** [comment]
any updated on this ? errors persits even with if `scan_params` in not null..

```python
File: crash-20170320-195149-kanaan-check_for_s3.c0-d0c7199b-56ac-47ec-8e34-fda4472f6051.pklz
Node: resting_preproc_HA040.func_gather_0.check_for_s3.c0
Working directory: /scr/sambesi4/workspace/project_cpac/preproc/working/resting_preproc_HA040/func_gather_0/_scan_scan_parameters/check_for_s3


Node inputs:

creds_path = None
file_path = {'acquisition': 'alt+z', 'tr': '2.4', 'last_tr': 'None', 'reference': '100', 'first_tr': 'None'}
function_str = def check_for_s3(file_path, creds_path, dl_dir=None, img_type='anat'):
    '''
    '''

    # Import packages
    import os
    import nibabel as nib
    import botocore.exceptions

    from indi_aws import fetch_creds

    # Init variables
    s3_str = 's3://'
    if dl_dir is None:
        dl_dir = os.getcwd()

    # Explicitly lower-case the "s3"
    if file_path.lower().startswith(s3_str):
        file_path_sp = file_path.split('/')
        file_path_sp[0] = file_path_sp[0].lower()
        file_path = '/'.join(file_path_sp)

    # Check for s3 string in filepaths
    if file_path.startswith(s3_str):
        # Get bucket name and bucket object
        bucket_name = file_path.replace(s3_str, '').split('/')[0]
        bucket = fetch_creds.return_bucket(creds_path, bucket_name)

        # Extract relative key path from bucket and local path
        s3_prefix = os.path.join(s3_str, bucket_name)
        s3_key = file_path.replace(s3_prefix, '').lstrip('/')
        local_path = os.path.join(dl_dir, os.path.basename(s3_key))

        # Get local directory and create folders if they dont exist
        local_dir = os.path.dirname(local_path)
        if not os.path.exists(local_dir):
            os.makedirs(local_dir)

        # Download file
        try:
            bucket.download_file(Key=s3_key, Filename=local_path)
        except botocore.exceptions.ClientError as exc:
            error_code = int(exc.response['Error']['Code'])
            if error_code == 403:
                err_msg = 'Access to bucket: "%s" is denied; using credentials '\
                          'in subject list: "%s"; cannot access the file "%s"'\
                          % (bucket_name, creds_path, file_path)
                raise Exception(err_msg)
            elif error_code == 404:
                err_msg = 'Bucket: "%s" does not exist; check spelling and try '\
                          'again' % bucket_name
                raise Exception(err_msg)
            else:
                err_msg = 'Unable to connect to bucket: "%s". Error message:\n%s'\
                          % (bucket_name, exc)
        except Exception as exc:
            err_msg = 'Unable to connect to bucket: "%s". Error message:\n%s'\
                      % (bucket_name, exc)
            raise Exception(err_msg)

    # Otherwise just return what was passed in
    else:
        local_path = file_path

    # Check image dimensionality
    img_nii = nib.load(local_path)
    if img_type == 'anat':
        if len(img_nii.shape) != 3:
            raise IOError('File: %s must be an anatomical image with 3 '\
                          'dimensions but %d dimensions found!' % (local_path,len(img_nii.shape)))
    elif img_type == 'func':
        if len(img_nii.shape) != 4:
            raise IOError('File: %s must be a functional image with 4 '\
                          'dimensions but %d dimensions found!' % (local_path,len(img_nii.shape)))

    # Return the local path
    return local_path

ignore_exception = False
img_type = func



Traceback:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nipype/pipeline/plugins/multiproc.py", line 49, in run_node
    result['result'] = node.run(updatehash=updatehash)
  File "/usr/lib/python2.7/dist-packages/nipype/pipeline/engine/nodes.py", line 394, in run
    self._run_interface()
  File "/usr/lib/python2.7/dist-packages/nipype/pipeline/engine/nodes.py", line 504, in _run_interface
    self._result = self._run_command(execute)
  File "/usr/lib/python2.7/dist-packages/nipype/pipeline/engine/nodes.py", line 630, in _run_command
    result = self._interface.run()
  File "/usr/lib/python2.7/dist-packages/nipype/interfaces/base.py", line 1043, in run
    runtime = self._run_wrapper(runtime)
  File "/usr/lib/python2.7/dist-packages/nipype/interfaces/base.py", line 1000, in _run_wrapper
    runtime = self._run_interface(runtime)
  File "/usr/lib/python2.7/dist-packages/nipype/interfaces/utility.py", line 499, in _run_interface
    out = function_handle(**args)
  File "<string>", line 18, in check_for_s3
AttributeError: 'dict' object has no attribute 'lower'
Interface Function failed to run.
```

**Comment by sgiavasis** [comment]
Hello,

We are putting priority on this one.

**Comment by sgiavasis** [comment]
This will be fixed in 1.0.1b.
